barretenberg/cpp/src/barretenberg/vm/avm_trace/avm_alu_trace.cpp

```diff
@@ -731,4 +743,218 @@ FF AvmAluTraceBuilder::op_cast(FF const& a, AvmMemoryTag in_tag, uint32_t clk)
     return c;
 }
 
+// Returns the number of bits associated with a given memory tag
+uint8_t mem_tag_bits(AvmMemoryTag in_tag)
+{
+    switch (in_tag) {
+    case AvmMemoryTag::U8:
+        return 8;
+    case AvmMemoryTag::U16:
+        return 16;
+    case AvmMemoryTag::U32:
+        return 32;
+    case AvmMemoryTag::U64:
+        return 64;
+    case AvmMemoryTag::U128:
+        return 128;
+    case AvmMemoryTag::FF:
```

Comment:
As the tag FF is not used, I would rather perform some error handling.

Reply:
I updated with an ASSERT for now. We might need to revisit some, more graceful error handling for this and the `op_not` version of this.